src/yb/util/wait_state.cc

```diff
@@ -66,6 +66,32 @@ WaitStateCode WaitStateInfo::get_frozen_state() const {
   return code_;
 }
 
+void WaitStateInfo::set_state_if(WaitStateCode prev, WaitStateCode c) {
```

Comment:
How does this work? I don't see the wait state code getting updated, or is it not supposed to do that at all?